src/main/java/org/squiddev/plethora/gameplay/modules/glasses/methods/MethodsCanvas3D.java

```diff
@@ -88,6 +86,34 @@ private MethodsCanvas3D() {
 		return baked.makeChild(box, canvas.reference(box)).getObject();
 	}
 
+	@PlethoraMethod(worldThread = false,
+		doc = "function(startX:number, startY:number, startZ:number, endX:number, endY:number, endZ:number[, thickness:number, color:number] -- Create a new line."
+	)
+	public static TypedLuaObject<Line3D> addLine(IContext<Group3D> baked, @FromContext CanvasServer canvas, Object[] args) throws LuaException {
+		double startX = getFloat(args, 0);
```

Comment:
You should be able to do:

```java
public static TypedLuaObject<Line3D> addLine(
    IContext<Group3D> baked, @FromContext CanvasServer canvas, 
    Vec3 start, Vec3d end, 
    @Optional(defFloat = 1.0) float thickness, @Optional(defInf = DEFAULT_COLOUR) int colour
) throws LuaException {
```

(maybe tweak the formatting there - no clue how this'll look in an editor).

Yes, I know this is slightly different, as you have to pass in a table instead of pure coordinates, but that's consistent with the 2D version.

This also means you can leave off the type from the doc comment. See the other functions for an example.